framework/wazuh/core/batcher/batcher.py

```diff
@@ -0,0 +1,196 @@
+
+
+class Batcher:
+    """
+    Manages the batching of messages from a MuxDemuxQueue and sends them in bulk to an indexer.
+
+    Parameters
+    ----------
+    queue : MuxDemuxQueue
+        The MuxDemuxQueue instance for message multiplexing and demultiplexing.
+    config : BatcherConfig
+        Configuration parameters for batching, such as maximum elements and size.
+    """
+    def __init__(self, queue: MuxDemuxQueue, config: BatcherConfig):
+        self.q: MuxDemuxQueue = queue
+        self._buffer: Buffer = Buffer(max_elements=config.max_elements, max_size=config.max_size)
+        self._timer: TimerManager = TimerManager(max_time_seconds=config.max_time_seconds)
+        self._shutdown_event: Optional[asyncio.Event] = None
+
+    async def _get_from_queue(self) -> Message:
+        """
+        Retrieves a message from the mux queue. If the queue is empty, waits and retries until a message is received
+        or the task is cancelled.
+
+        Returns
+        -------
+        Message
+            The message retrieved from the mux queue.
+
+        Raises
+        ------
+        asyncio.CancelledError
+            If the task is cancelled during the operation.
+        """
+        message = None
+        while True:
+            try:
+                message = self.q.receive_from_mux(block=False)
+                return message
+            except queue.Empty:
+                await asyncio.sleep(WAIT_TIME_BETWEEN_QUEUE_READ)
+            except asyncio.CancelledError:
+                if message is not None:
+                    self.q.send_to_mux(uid=message.uid, msg=message.msg)
+                break
+
+    async def _send_buffer(self, events: List[Message]):
+        """
+        Sends a batch of messages to the indexer in bulk. Updates the demux queue with the response messages.
+
+        Parameters
+        ----------
+        events : List[Message]
+            The list of messages to be sent in bulk.
+
+        Raises
+        ------
+        Exception
+            If an error occurs during the sending of the buffer.
+        """
+        try:
+            async with get_indexer_client() as indexer_client:
+                list_of_uid: List[uuid.UUID] = [event.uid for event in events]
+                bulk_list: List[BulkDoc] = [
+                    BulkDoc.create(index=indexer_client.events.INDEX, doc_id=None, doc=event.msg) for event in events
+                ]
+
+                response = await indexer_client.events.bulk(data=bulk_list)
+
+                for response_item, uid in zip(response["items"], list_of_uid):
+                    response_item_msg = response_item["create"]
+                    response_msg = Message(uid=uid, msg=response_item_msg)
+                    self.q.send_to_demux(response_msg)
+        except Exception as e:
+            tb_str = traceback.format_exception(etype=type(e), value=e, tb=e.__traceback__)
+            logger.error(f"Error when sending buffer:\n{''.join(tb_str)}")
```

Comment:
We should avoid multi-line errors, they mess up the logs and it's hard to debug them later. I would prefer using a code so the user can get more information elsewhere

```suggestion
            logger.error(f"Error sending message to buffer: {''.join(tb_str)}")
```

Reply:
Changed in https://github.com/wazuh/wazuh/commit/7344dcea68aa1f5d451757e21ea6e02cc5e61448.